pkg/reconciler/configuration/configuration.go

```diff
@@ -192,10 +186,93 @@ func (c *Reconciler) reconcile(ctx context.Context, config *v1alpha1.Configurati
 	default:
 		return fmt.Errorf("unrecognized condition status: %v on revision %q", rc.Status, revName)
 	}
+	if rc != nil && rc.Status == corev1.ConditionTrue {
+		old, new := config.Status.LatestReadyRevisionName, lcr.Name
+		config.Status.SetLatestReadyRevisionName(lcr.Name)
+		if old != new {
+			c.Recorder.Eventf(config, corev1.EventTypeNormal, "LatestReadyUpdate",
+				"LatestReadyRevisionName updated to %q", new)
+		}
+	} else {
+		if err = c.findAndSetLatestReadyRevision(config); err != nil {
+			return fmt.Errorf("failed to find and set latest ready revision: %w", err)
+		}
+	}
 
 	return nil
 }
 
+// findAndSetLatestReadyRevision finds the last ready revision and sets LatestReadyRevisionName to it
+func (c *Reconciler) findAndSetLatestReadyRevision(config *v1alpha1.Configuration) error {
+	sortedRevisions, err := c.getSortedCreatedRevisions(config)
+	if err != nil {
+		return err
+	}
+	for _, rev := range sortedRevisions {
+		if rev.Status.IsReady() {
+			// No need to update latest ready revision in this case
+			if rev.Name == config.Status.LatestReadyRevisionName {
+				return nil
+			}
+			config.Status.SetLatestReadyRevisionName(rev.Name)
+			c.Recorder.Eventf(config, corev1.EventTypeNormal, "LatestReadyUpdate",
+				"LatestReadyRevisionName updated to %q", rev.Name)
+			return nil
+		}
+	}
+	return nil
+}
+
+// getSortedCreatedRevisions returns the list of created revisions sorted in descending
+// generation order between the generation of the latest ready revision and config's generation.
+func (c *Reconciler) getSortedCreatedRevisions(config *v1alpha1.Configuration) ([]*v1alpha1.Revision, error) {
+	lister := c.revisionLister.Revisions(config.Namespace)
+	configSelector := labels.SelectorFromSet(map[string]string{
+		serving.ConfigurationLabelKey: config.Name,
+	})
+	if config.Status.LatestReadyRevisionName != "" {
+		latestReadyRev, err := lister.Get(config.Status.LatestReadyRevisionName)
+		if err != nil {
+			return nil, err
+		}
+		start := latestReadyRev.Generation
+		var generations []string
+		for i := start + 1; i <= int64(config.Generation); i++ {
+			generations = append(generations, strconv.FormatInt(i, 10))
+		}
+
+		// Add an "In" filter so that the configurations we get back from List have generation
+		// in range (config's latest ready generation, config's generation]
+		generationKey := serving.ConfigurationGenerationLabelKey
+		inReq, err := labels.NewRequirement(generationKey,
+			selection.In,
+			generations,
+		)
+		if err != nil {
+			return nil, err
+		}
+		configSelector = configSelector.Add(*inReq)
+	}
+
+	list, err := lister.List(configSelector)
+
+	if err == nil && len(list) > 0 {
+		// Return a sorted list with Generation in descending order
+		if len(list) > 1 {
```

Comment:
This double predicate is a bit weird, I would use:
```
if err != nil {
  return nil, err
}
// Does this panic if list is nil? I doubt it would...
sort.Slice(list, func(){})
```